src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -0,0 +1,67 @@
+package wolox.training.controllers;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import wolox.training.exceptions.books.BookNotFoundException;
+import wolox.training.exceptions.users.UserNotFoundException;
+import wolox.training.models.Book;
+import wolox.training.models.User;
+import wolox.training.repositories.UserRepository;
+
+@RestController
+@RequestMapping("/api/users")
+public class UserController {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @GetMapping("/username/{username}")
+    public User findByUsername(@PathVariable String username) {
+        return userRepository.findByUsername(username)
+                .orElseThrow(() -> new UserNotFoundException(
+                        "User with username " + username + " not found"));
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public User create(@RequestBody User user) {
+        return userRepository.save(user);
+    }
+
+    @GetMapping
+    public List<User> getAll() {
+        List<User> users = new ArrayList<>();
+        userRepository.findAll().forEach(users::add);
```

Comment:
You can simply return the result of the method `userRepository.findAll()` :)

Reply:
![image](https://user-images.githubusercontent.com/19678474/80393090-9fb1c400-8875-11ea-9afa-c712c7446c74.png)

It doesn't work :c

Reply:
If you want you can try with `Lists.newArrayList(userRepository.findAll())` 😄 
Check this out: https://www.baeldung.com/java-iterable-to-collection#guava-1